test/BuildingRegistry.Console.ReactivatedHouseNumbers/Program.cs

```diff
@@ -0,0 +1,104 @@
+namespace BuildingRegistry.Console.ReactivatedHouseNumbers
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using Amazon.DynamoDBv2;
+    using Api.CrabImport.CrabImport.Requests;
+    using Be.Vlaanderen.Basisregisters.Crab;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Import.Processing.Json;
+    using Building.Commands.Crab;
+    using Building.Events.Crab;
+    using Dapper;
+    using Microsoft.Data.SqlClient;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using NodaTime;
+
+
+    //Case 5f: when housenumber relation is retired then becomes active again (correction), buildings with more than 2 subaddresses don't have the right amount of units
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var configuration = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                .AddJsonFile($"appsettings.{Environment.MachineName.ToLowerInvariant()}.json", optional: true, reloadOnChange: false)
+                .AddEnvironmentVariables()
+                .AddCommandLine(args ?? new string[0])
+                .Build();
+
+            var eventsConnectionString = configuration.GetConnectionString("Events");
+            var jsonSerializerSettings = new JsonSerializerSettings().ConfigureDefaultForEvents();
+            int count = 0;
+            var lastStreamId = -1;
+
+            // GET STREAMS
+            using (var sqlConnection = new SqlConnection(eventsConnectionString))
+            {
+                var streamsIds = sqlConnection.Query<int>(
+                    @"SELECT TOP(10000) IdInternal FROM [building-registry-events].[BuildingRegistry].[Streams] WHERE IdInternal > @lastStreamId ORDER BY IdInternal", new { lastStreamId })
+                    .ToList();
+                do
+                {
+                    foreach (var streamId in streamsIds)
+                    {
+                        var importedTerrainObjectHouseNumbers = sqlConnection.Query<string>(
+                            @"SELECT JsonData FROM [building-registry-events].BuildingRegistry.Messages
+                            WHERE StreamIdInternal = @streamId AND [type] = 'TerrainObjectHouseNumberWasImportedFromCrab'
+                            ORDER BY Position", new { streamId })
+                            .Select(x => JsonConvert.DeserializeObject<TerrainObjectHouseNumberWasImportedFromCrab>(x, jsonSerializerSettings))
+                            .ToList();
+
+                        foreach (var terrainObjectHouseNumberWasImportedFromCrab in importedTerrainObjectHouseNumbers.GroupBy(x => x.TerrainObjectHouseNumberId))
+                        {
+                            if(terrainObjectHouseNumberWasImportedFromCrab.Any(x => x.Modification == CrabModification.Delete))
```

Comment:
Playing devil's advocate, is it possible after a `Delete` it gets _revived_ again?

Reply:
No, that's impossible